### DISCUSSION TITLE
LSD2: Trouble with Branch Collapsing in IQ-TREE: Seeking Clarification on -l Option

### DISCUSSION THREAD

**Discussion by YangZhuoya** [iqtree/iqtree2#187] [Q&A] [answered]
I am currently using IQ-TREE version 2.3.2 for phylogenetic analysis and encountering an issue where a significant number of internal branches are automatically collapsed despite setting extremely low branch length thresholds. The error messages suggest using the -l option to set branch lengths, but this option appears to be invalid. I tried other versions and none of them works. Does someone know if the -l option is available or how can I  generate binary tree without branch collapsing?

**Comment by roblanf** [comment]
Hi @YangZhuoya,

Can you post your commandline and the output file with the collapsed branches?

Thanks,

Rob

**Reply by YangZhuoya** [reply]
Hi Rob,

Thank you for your response. Below is the command line I used in IQ-TREE:

iqtree -s CA_S450L_300.fasta -keep-ident -m GTR+F+G4 --date CA_S450L_date.txt -fconst 758047,1449684,1444318,757918

And here is the output log file [CA_S450L_300.fasta.log](https://github.com/iqtree/iqtree2/files/15194852/CA_S450L_300.fasta.log). 

It said "Collapse 110 (over 297) internal branches having branch length <= 1.13e-07 (settable via option -l)". I look forward to any insights you might have on preventing these branches from collapsing. Thanks!

Zhuoya

**Reply by bqminh** [reply]
Please check if this thread in google group answers your question:
https://groups.google.com/g/iqtree/c/KVJ9Tx7YvRE/m/Lu4Gkpy-AAAJ
If not, please ask the author of LSD2 directly: https://github.com/tothuhien/lsd2. IQ-TREE merely uses LSD2 for tree dating.

**Reply by YangZhuoya** [reply]
Hi Rob
I found the solutions. Thanks so much!
Best,
Zhuoya

**Comment by tothuhien** [comment, marked as answer]
Hi Zhuoya,
I just answered the same question here:
https://github.com/iqtree/iqtree2/discussions/189
So briefly, if you want of force LSD2 not to make polytomies, for example you want the minimum branch length is 0.001, you can add the option:
--date-options "-l 0 -u 0.001"
Cheers,
Hien

**Reply by YangZhuoya** [reply]
Hi hien,
Problem solved! Thanks so much for the help!
Best,
Zhuoya
